src/core/Akka.Remote/Transport/DotNetty/BatchWriter.cs

```diff
@@ -134,52 +144,50 @@ public override Task WriteAsync(IChannelHandlerContext context, object message)
             return write;
         }
 
+        public override void Flush(IChannelHandlerContext context)
+        {
+            // reset statistics upon flush
+            Reset();
+            base.Flush(context);
+        }
+
+        public override void UserEventTriggered(IChannelHandlerContext context, object evt)
+        {
+            if (evt is FlushCmd)
+            {
+                if (HasPendingWrites)
+                {
+                    context.Flush();
+                    Reset();
+                }
+            }
+        }
+
         public override Task CloseAsync(IChannelHandlerContext context)
         {
             // flush any pending writes first
             context.Flush();
+            _flushSchedule?.Cancel();
             CanSchedule = false;
             return base.CloseAsync(context);
         }
 
         private void ScheduleFlush(IChannelHandlerContext context)
         {
             // Schedule a recurring flush - only fires when there's writable data
-            var task = new FlushTask(context, Settings.FlushInterval, this);
-            context.Executor.Schedule(task, Settings.FlushInterval);
+            _flushSchedule = Scheduler.Advanced.ScheduleRepeatedlyCancelable(Settings.FlushInterval,
+                Settings.FlushInterval,
+                () =>
+                {
+                    // want to fire this event through the top of the pipeline
+                    context.Channel.Pipeline.FireUserEventTriggered(FlushCmd.Instance);
```

Comment:
I wanted to use an event for this as this will allow me to do some sampling around how often the `FlushCmd` has to be invoked separately from other flushes. But that may end up being a distinction without difference depending on how I write the sampler.